### ISSUE TITLE
Batchnorm layer with scale_bias: true

### ISSUE THREAD

**Issue by shuyu0815** [NVIDIA/caffe#394]
I tried to implement inference with trained model (not trained with nvcaffe) in nvcaffe0.16.
I try to modify the prototxt to meet the prototxt used by nvcaffe=0.16

Because the trained model has the Scale layer and I set scale_bias: true 
but I got an error:

I0810 08:07:56.287736 13150 net.cpp:405] Network initialization done.
I0810 08:07:56.329692 13150 net.cpp:1077] Copying source layer data Type:Data #blobs=0
I0810 08:07:56.329742 13150 net.cpp:1077] Copying source layer conv1 Type:Convolution #blobs=2
F0810 08:07:56.329902 13150 net.cpp:1075] Check failed: target_blobs.size() == source_layer.blobs_size() (5 vs. 3) Incompatible number of blobs for layer bn_conv1

It seems that the BatchNorm layer can't catch scale's parameters automatically !
Anyone can give me some suggestions or show some example?
Thanks !

The following is the modified prototxt and I directly replace Scale layer by setting scale_bias true:

layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  convolution_param {
    num_output: 32
    kernel_size: 5
    stride: 2
    pad: 2
    weight_filler {
      type: "xavier"
    }
  }
}
layer {
  name: "bn_conv1"
  type: "BatchNorm"
  bottom: "conv1"
  top: "bn_conv1"
  batch_norm_param {
    use_global_stats: true
    scale_bias: true
  }
}
layer {
  name: "relu_conv1"
  type: "ReLU"
  bottom: "bn_conv1"
  top: "bn_conv1"
}

Thanks for help !

**Comment by drnikolaev** [comment]
Hi @shuyu0815 we changed implementation so that if `scale_bias: true` then we have 2 extra blobs (thus total is 3+2=5). Please re-train in NVCaffe 0.16, this is the only way.
